streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignorTest.java

```diff
@@ -798,17 +798,43 @@ public void shouldReturnEmptyClusterMetadataIfItHasntBeenBuilt() throws Exceptio
     }
 
     @Test
-    public void shouldNotLoopInfinitelyOnMissingMetadataAndShouldNotCreateRelatedTasks() {
+    public void shouldCreateAllTasksAndInternalTopicIfAllInputTopicsAreKnown() {
+        runMetadataTest("topic1", true, "topic3", true);
+    }
+
+    @Test
+    public void shouldNotLoopInfinitelyOnPartlyMissingMetadataAndShouldNotCreateRelatedTasks1() {
+        runMetadataTest("topic1", true, "unknownTopic", false);
+    }
+
+    @Test
+    public void shouldNotLoopInfinitelyOnPartlyMissingMetadataAndShouldNotCreateRelatedTasks2() {
+        runMetadataTest("unknownTopic", false, "topic3", true);
+    }
+
+    @Test
+    public void shouldNotLoopInfinitelyOnMissingMetadataForAllInputTopicsAndShouldNotCreateRelatedTasks() {
+        runMetadataTest("unknownTopic1", false, "unknownTopic2", false);
+    }
+
+    private void runMetadataTest(
```

Comment:
Sure. I thought the actual test name should be sufficient. It's just a helper not a full test. :)